### PR TITLE
Add supplementary documentation about how to use optimizers

### PR DESCRIPTION
**Pull request by shendiaomo**
Supplementary documentation for #1376

### CODE REVIEW

File: doc/tutorial/iris-dnn.md [hunk 1]
```diff
@@ -50,17 +48,75 @@ Putting it all together, we have our first SQLFlow training statement.
...
   model.n_classes = 3,
   model.hidden_units = [10, 10],
-  train.epoch = 100
+  train.epoch = 10
+COLUMN sepal_length, sepal_width, petal_length, petal_width
+LABEL class
+INTO sqlflow_models.my_dnn_model;
+```
+
+Under normal conditions, the training statement will complete after several minutes.
```
Comment (wangkuiyi) [inline doc/tutorial/iris-dnn.md]: The above training statement usually takes a few minutes to run, and the outputs look like the following:

```python
{'accuracy': 0.4, 'average_loss': 1.0920922, 'loss': 1.0920922, 'global_step': 1100}
```

File: doc/tutorial/iris-dnn.md [hunk 2]
```diff
@@ -50,17 +48,75 @@ Putting it all together, we have our first SQLFlow training statement.
...
+
+## Tune
+
+Typically, the above training statement will output something like:
+
+```python
+{'accuracy': 0.4, 'average_loss': 1.0920922, 'loss': 1.0920922, 'global_step': 1100}
+
+```
+which seems to be not ideal.
```
Comment (wangkuiyi) [inline doc/tutorial/iris-dnn.md]: The average loss doesn't look very good; an idea value should about .....  Let us see what we can do to improve model quality.

File: doc/tutorial/iris-dnn.md [hunk 3]
```diff
@@ -50,17 +48,75 @@ Putting it all together, we have our first SQLFlow training statement.
...
+
+```
+which seems to be not ideal. 
+
+In order to improve the model performance, we can tune the [hyperparameters](https://en.wikipedia.org/wiki/Hyperparameter_(machine_learning)) manually.
+> In machine learning, a hyperparameter is a parameter whose value is set before the learning process begins. By contrast, the values of other parameters are derived via training.
+
+According to the [Universal approximation theorem](https://en.wikipedia.org/wiki/Universal_approximation_theorem), the architecture of a multilayer [feed-forward network](https://en.wikipedia.org/wiki/Feedforward_neural_network) (such as our `DNNClassifier`) gives the neural network the potential of being a universal approximator.
+
+Our first _performance improvement plan_ is to tune the architecture of our model by increasing the `hidden_units` of each layer to 100 because the width of feed-forward networks matters in the theorem.
```
Comment (wangkuiyi) [inline doc/tutorial/iris-dnn.md]: _performance improvement plan_ => trial

File: doc/tutorial/iris-dnn.md [hunk 4]
```diff
@@ -50,17 +48,75 @@ Putting it all together, we have our first SQLFlow training statement.
...
 INTO sqlflow_models.my_dnn_model;
 ```
+The above statement will give a decent result like:
+
+```python
+{'accuracy': 0.98, 'average_loss': 0.10286382, 'loss': 0.10286382, 'global_step': 1100}
+
+```
+
+That's all you need to know about tuning in this tutorial. In fact, tuning is the most important and most time-consuming work of a machine learning engineer. Adjusting hyperparameters in a production environment is quite subtle.
```
Comment (wangkuiyi) [inline doc/tutorial/iris-dnn.md]: Tuning usually takes a large fraction of the working hours of data scientists.  The SQLFlow team has a plan to introduce auto hyperparameter tuning in the near future.

File: doc/tutorial/iris-dnn.md (hunk 2 above)
Comment (shendiaomo) [reply doc/tutorial/iris-dnn.md]: Done.

File: doc/tutorial/iris-dnn.md (hunk 3 above)
Comment (shendiaomo) [reply doc/tutorial/iris-dnn.md]: Done.

File: doc/tutorial/iris-dnn.md (hunk 4 above)
Comment (shendiaomo) [reply doc/tutorial/iris-dnn.md]: Done.

**Review by wangkuiyi — APPROVED** [review]
